src/restore.ts

```diff
@@ -34,7 +35,16 @@ async function run() {
 
     core.info(`... Restoring cache ...`);
     const key = config.cacheKey;
-    const restoreKey = await cache.restoreCache(config.cachePaths, key, [config.restoreKey]);
+    const restoreKey = await withRetries(
+      () =>
+        withTimeout(
+          () => cache.restoreCache(config.cachePaths, key, [config.restoreKey]),
+          config.timeout
+        ),
+      config.maxRetryAttempts,
+      () => true
+    );
```

Comment:
The composition is nice, but the construct still looks very unweildy looking at it, I wonder if you can package that up into a single fn?